### PR TITLE
NO AUTO Renames core to common package, makes core and full depend on it

### PR DESCRIPTION
**Pull request by ncordon**
## What
Renames core to common, creates a new core package and makes core and full depend on common.

## Why
This is a step towards splitting core and full completely. Next step should move procedures from common to core, increasingly.

## Next
- [ ] Processor dependency in `core`? At the moment it's left in `common`
- [ ] `core->test` depends on `common->test`. Should we remove that in the future?
- [ ] Fix some procedures / functions not being available in `full`.

### CODE REVIEW

File: full/src/test/java/apoc/ApocSplitTest.java [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+        "apoc.es.get",
+        "apoc.es.query",
+        "apoc.es.getRaw",
+        "apoc.es.postRaw",
+        "apoc.es.post",
+        "apoc.es.put",
+//        "apoc.export.xls.all",
+//        "apoc.export.xls.data",
+//        "apoc.export.xls.graph",
+//        "apoc.export.xls.query",
```
Comment (ncordon) [inline full/src/test/java/apoc/ApocSplitTest.java:569]: I think there's something broken cause this should be in full, and it's packaged in the jar but it's not available in the procedures. Way to reproduce:

```sh
docker run --restart always --publish=7474:7474 --publish=7687:7687 --env=NEO4J_ACCEPT_LICENSE_AGREEMENT=yes --env NEO4JLABS_PLUGINS='["apoc"]' neo4j:4.4.9-enterprise
```

```
// In cypher-shell

neo4j@neo4j> RETURN apoc.version();
+----------------+
| apoc.version() |
+----------------+
| "4.4.0.7"      |
+----------------+

1 row
ready to start consuming query after 75 ms, results consumed after another 2 ms
neo4j@neo4j> CALL apoc.export.xls.all
             ;
There is no procedure with the name `apoc.export.xls.all` registered for this database instance. Please ensure you've spelled the procedure name correctly and that the procedure is properly deployed.
```

File: full/src/test/java/apoc/ApocSplitTest.java [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+        "apoc.cypher.parallel2",
+        "apoc.gephi.add",
+        "apoc.mongo.aggregate",
+        "apoc.mongo.count",
+        "apoc.mongo.find",
+        "apoc.mongo.insert",
+        "apoc.mongo.update",
+        "apoc.mongo.delete",
+        "apoc.mongodb.get.byObjectId",
+        //"apoc.load.xls",
```
Comment (ncordon) [inline full/src/test/java/apoc/ApocSplitTest.java:530]: See [this](https://github.com/neo4j-contrib/neo4j-apoc-procedures/pull/3079/commits/53da905ca870041d26a60c966f3c3e7803ce164f#r933362761) comment

File: full/src/test/java/apoc/ApocSplitTest.java [hunk 3]
```diff
@@ -0,0 +1,631 @@
...
+    public static final List<String> FULL_FUNCTIONS = List.of(
+        "apoc.trigger.nodesByLabel",
+        "apoc.trigger.propertiesByKey",
+        "apoc.trigger.toNode",
+        "apoc.trigger.toRelationship",
+        "apoc.ttl.config",
+        "apoc.static.get",
+        "apoc.static.getAll",
+        "apoc.coll.avgDuration"
+        //"apoc.data.email"
```
Comment (ncordon) [inline full/src/test/java/apoc/ApocSplitTest.java:606]: See [this](https://github.com/neo4j-contrib/neo4j-apoc-procedures/pull/3079/commits/53da905ca870041d26a60c966f3c3e7803ce164f#r933362761) comment

File: full/src/test/java/apoc/ApocSplitTest.java [hunk 4]
```diff
@@ -0,0 +1,631 @@
...
+import java.util.stream.Stream;
+import org.neo4j.driver.Session;
+
+import static org.junit.Assert.assertTrue;
+
+/*
+ This test is just to verify the split of core and full
+ */
+public class ApocSplitTest {
+    public static final List<String> PROCEDURES_FROM_CORE = List.of(
```
Comment (ncordon) [inline full/src/test/java/apoc/ApocSplitTest.java:19]: The aim is that this list should be empty at the end of the process

File: full/src/test/java/apoc/ApocSplitTest.java [hunk 5]
```diff
@@ -0,0 +1,631 @@
...
+        "apoc.convert.setJsonProperty",
+        "apoc.convert.toTree",
+        "apoc.neighbors.tohop",
+        "apoc.neighbors.tohop.count",
+        "apoc.neighbors.byhop",
+        "apoc.neighbors.byhop.count",
+        "apoc.neighbors.athop",
+        "apoc.neighbors.athop.count");
+
+    public static final List<String> FUNCTIONS_FROM_CORE = List.of(
```
Comment (ncordon) [inline full/src/test/java/apoc/ApocSplitTest.java:206]: The aim is that this list should be empty at the end of the process

File: core/src/test/java/apoc/ApocSplitTest.java [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+        "apoc.convert.setJsonProperty",
+        "apoc.convert.toTree",
+        "apoc.neighbors.tohop",
+        "apoc.neighbors.tohop.count",
+        "apoc.neighbors.byhop",
+        "apoc.neighbors.byhop.count",
+        "apoc.neighbors.athop",
+        "apoc.neighbors.athop.count");
+
+    public static final Set<String> CORE_FUNCTIONS = Set.of("apoc.temporal.format",
```
Comment (AzuObs) [inline core/src/test/java/apoc/ApocSplitTest.java:203]: Small detail but maybe "apoc.temporal.format" could be on new line as it is above

File: full/src/test/java/apoc/ApocSplitTest.java (hunk 1 above)
Comment (AzuObs) [reply full/src/test/java/apoc/ApocSplitTest.java:569]: Oki that should be fine. Maybe we could add a todo to make sure it doesn't get forgotten about?

File: common/build.gradle [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    tasks.withType(GroovyCompile) {
+      groovyOptions.fork = false
+    }
+    tasks.withType(Test) {
+      // containers (currently) have 2 dedicated cores and 4GB of memory
+      maxParallelForks = 2
+      minHeapSize = '128m'
+    }
+  }
+}
```
Comment (AzuObs) [inline common/build.gradle:169]: Do you think we could remove this completely in the 3 common/core/processor build.gradle files? 

I'm not sure the tests are particularly parallelizable unfortunately, even when run locally.

**Review by AzuObs — COMMENTED** [review]
Really nice work so far. I had a couple of small suggestions.

I also have a question regarding the core tests. It seems like those tests are currently not run, or at least I couldn't find any evidence that they were. Do you think that is indeed the case?

File: common/build.gradle (hunk 7 above)
Comment (ncordon) [reply common/build.gradle]: I've removed it, so let's see, but I'm not a fan of introducing unnecessary changes in PRs and this was pre-existing

**Comment by ncordon** [comment]
> I also have a question regarding the core tests. It seems like those tests are currently not run, or at least I couldn't find any evidence that they were

They should, but I forgot to archieve them

**Review by AzuObs — COMMENTED** [review]
Great job 🎉

**Review by AzuObs — APPROVED** [review]

**Pull request merged**